tests/rbd_mirror/test_image_operations_on_snap_mirroring.py

```diff
@@ -0,0 +1,160 @@
+                    "primary": kw.get("rbd_obj"),
+                }
+                for cluster_type, rbd in test_obj.items():
+                    log.info(
+                        f"Testing image expand/shrink/remove on {cluster_type} cluster for image {image}"
+                    )
+                    ret_val = test_image_remove_resize(
+                        rbd=rbd,
+                        pool=pool,
+                        image=image,
+                        image_size=image_size,
+                        cluster_type=cluster_type,
+                    )
+                    if ret_val:
+                        return 1
+                return 0
+    return 0
+
+
+def run(**kw):
+    """Verify image operations from primary and secondary cluster
+    Pre-requisites :
+    We need atleast one client node with ceph-common, fio and rbd-nbd packages,
+    conf and keyring files in both clusters with snapshot based RBD mirroring
+    enabled between the clusters.
+    Test cases covered -
+    1) CEPH-83574857 - Configure two-way rbd-mirror (replicated and ec pool) on
+    Stand alone CEPH cluster on image with snapshot based mirroring and perform
+    image operations like expand/shrink/remove from primary and secondary clusters.
+    kw:
+        config:
+            rep_pool_config:
+              num_pools: 1
+              num_images: 10
+              mode: pool
+              mirrormode: snapshot
+            ec_pool_config:
+              num_pools: 1
+              num_images: 10
+              mode: pool
+              mirrormode: snapshot
+
+    Test Case Flow
+    1. Bootstrap two CEPH clusters and create a pool and 10 images each for replicated and ec pools
+    2. Setup snapshot based mirroring in between these clusters.
+    3. Run IOs on the Primary cluster for the image.
+    4. Expand the image while IO is in progress in primary cluster
+    5. Shrink the image while IO is in progress in primary cluster
+    6. Remove the image while IO is in progress in primary cluster
+    7. Try expand/shrink/remove operation from secondary cluster
+    """
+    pool_types = ["rep_pool_config", "ec_pool_config"]
+    log.info(
+        "CEPH-83574857 - Running verify image operations from primary and secondary cluster"
+    )
+    try:
+        ret_val = 0
+        kw.get("config", {})["do_not_run_io"] = True
+        mirror_obj = initial_mirror_config(**kw)
+        cluster_name = kw["ceph_cluster"].name
+        sec_cluster_name = [
+            k for k in kw.get("ceph_cluster_dict").keys() if k != cluster_name
+        ][0]
+        pool_types = mirror_obj.get(cluster_name).get("pool_types")
+        rbd_obj = mirror_obj.get(cluster_name).get("rbd")
+        sec_obj = mirror_obj.get(sec_cluster_name).get("rbd")
+        ret_val = test_image_operations(
+            rbd_obj=rbd_obj, sec_obj=sec_obj, pool_types=pool_types, **kw
+        )
+        if ret_val:
+            log.error("Testing toggle journaling feature failed")
```

Comment:
looks like error message is not related to this test case.

Reply:
done.